src/parser.cpp

```diff
@@ -0,0 +1,70 @@
+#include <cstddef>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <iterator>
+#include <ostream>
+#include <stdlib.h>
+#include "../lib/simdjson.h"
+#include "parser.h"
+#include <string_view>
+#include <sys/stat.h>
+
+
+using namespace std;
+
+extern "C" {  
+
+  Program* allocate(const char *filename) {
+    // Allocate custom Program in memory
+    struct stat st;
+    stat(filename, &st);
+    size_t size = st.st_size;
+    return (Program *)malloc(size * sizeof(int));
+  }
+
+  void parseArray(const simdjson::dom::array elemArray, char **structArray) {
+    int elements = elemArray.size();
+    int max_chars = 100;
+    structArray = (char **)malloc(elements * max_chars * sizeof(char));
```

Comment:
Since the array will contain pointers to strings, shouldn't this be:
```suggestion
    structArray = (char **)malloc(elements * sizeof(char*));
```

Reply:
Yes, you are totally correct, i did not know how allocation worked so i simply put some extra memory. 
On last commit allocations have changes. 
Still not sure if done correctly.